assignment.rb

```diff
@@ -0,0 +1,295 @@
+
+    # Depending on the user's input, perform the correct function
+    case acc_menu_sel
+    # Account Menu option 1 indicates the user wants to check their balance.
+    when 1
+      puts "Your current account balance is: $#{@user.balance}"
+      puts ""
+    # Account Menu option 2 indicates the user would like to withdraw funds
+    when 2
+      # Make a withdrawal
+      withdraw_funds()
+    # Account Menu option 3 indicates the user would like to log out
+    when 3
+      @show_acc_menu = false
+      puts "Returning to the Main Menu."
+      puts ""
+    # If menu options 1-3 have failed, the world might end. Display an error
+    else
+      display_error()
+    end
+  end
+
+  # This method handles the withdrawing funds logic.
+  def withdraw_funds
+    # Ask the user how much they would like to withdraw from their account
+    amount = get_input("How much would you like to withdraw?").to_i
+    # Let's make sure the account and the ATM can handle this transaction
+    if @user.can_withdraw?(amount) && @atm.can_withdraw?(amount)
+      # Passed that check. Now update account amounts in memory and on file...
+      @user.deduct(amount)
+      @atm.deduct(amount)
+      update_data_file(@data_file)
+      # ... and dispense the monies.
+      puts "Sure thing. Dispensing cash below..."
+      puts "" 
+      puts "Your new balance is $#{@user.balance}"
+      puts ""
+    elsif amount <= 0
+      puts "You can not withdraw a negative amount."
+      puts ""
+    elsif !@user.can_withdraw?(amount)
+      puts "You have asked to withdraw more than you have in your account"
+      puts ""
+    elsif !@atm.can_withdraw?(amount)
+      puts "ATM has insufficient balance to fulfill this request."
+      puts ""
+    else
+      display_error()
+    end
+  end
+
+  # This method handles parsing the users.csv file that serves as our database table.
+  def parse_data_file(file_name)
+    CSV.foreach(file_name, headers: true) do |row|
+      user = User.new
+      user.name = row["Name"]
+      user.pin = row["Pin"]
+      user.balance = row["Balance"].to_i
+      @account_holders.push(user)
+    end
+    @account_holders
+  end
+
+  # This method handles updating our user data file after a transaction to ensure the
+  # file contains the most up to date information.
+  def update_data_file(file_name)
+    header = ["Name", "Pin", "Balance"]
+    CSV.open(file_name, "wb") do |csv|
+      csv << header
+      @account_holders.each do |user|
+        data = [user.name, user.pin, user.balance]
+        csv << data
+      end
+    end
+  end
+
+# Closes the class
+end
+
+
+
+
+
+# OK, lets initialize this ATM with a $10,000 balance
+atm = Machine.new
+atm.balance = 10000
+
+
+# This line creates a new instance of the Program.
+program = Program.new(atm)
+
+# This executes the program until the user has decided to exit.
+while program.exit_prog == false
+  program.run
+end
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
```

Comment:
No problemo... This was just so I wasn't stuck on the bottom in Sublime. Atom allows me to scroll past the last line though, so I shouldn't need to do this again.